sdk/samples/iot/hub/src/paho_iot_hub_pnp_component_example.c

```diff
@@ -0,0 +1,859 @@
+  }
+
+  mqtt_connect_options.ssl = &mqtt_ssl_options;
+
+  // Connect to IoT Hub
+  if ((rc = MQTTClient_connect(mqtt_client, &mqtt_connect_options)) != MQTTCLIENT_SUCCESS)
+  {
+    LOG_ERROR("Failed to connect, return code %d", rc);
+    exit(rc);
+  }
+
+  LOG_SUCCESS("Connected to IoT Hub");
+}
+
+static void subscribe(void)
+{
+  int rc;
+
+  // Subscribe to the commands topic. Messages received on this topic are commands to be invoked
+  // on the device.
+  if ((rc = MQTTClient_subscribe(mqtt_client, AZ_IOT_HUB_CLIENT_METHODS_SUBSCRIBE_TOPIC, 1))
+      != MQTTCLIENT_SUCCESS)
+  {
+    LOG_ERROR("Failed to subscribe to the commands subscribe topic filter, return code %d", rc);
+    exit(rc);
+  }
+
+  // Subscribe to the desired properties PATCH topic. Messages received on this topic will be
+  // updates to the desired properties.
+  if ((rc = MQTTClient_subscribe(mqtt_client, AZ_IOT_HUB_CLIENT_TWIN_PATCH_SUBSCRIBE_TOPIC, 1))
+      != MQTTCLIENT_SUCCESS)
+  {
+    LOG_ERROR("Failed to subscribe to the twin patch topic filter, return code %d", rc);
+    exit(rc);
+  }
+
+  // Subscribe to the twin response topic. Messages received on this topic will be response statuses
+  // from published reported properties or the requested twin document from twin GET publish
+  // messages.
+  if ((rc = MQTTClient_subscribe(mqtt_client, AZ_IOT_HUB_CLIENT_TWIN_RESPONSE_SUBSCRIBE_TOPIC, 1))
+      != MQTTCLIENT_SUCCESS)
+  {
+    LOG_ERROR("Failed to subscribe to twin response topic filter, return code %d", rc);
+    exit(rc);
+  }
+
+  LOG_SUCCESS("Subscribed to IoT Hub topics");
+}
+
+static void send_twin_get_message(void)
+{
+  int rc;
+
+  az_span request_id_span = get_request_id();
+  if (az_failed(
+          rc = az_iot_hub_client_twin_document_get_publish_topic(
+              &client, request_id_span, publish_message.topic, publish_message.topic_length, NULL)))
+  {
+    LOG_ERROR("Could not get twin get publish topic, az_result %d", rc);
+    exit(rc);
+  }
+
+  LOG_SUCCESS("Sending twin get request");
+  mqtt_publish_message(publish_message.topic, AZ_SPAN_NULL, 0);
+}
+
+// Send JSON formatted telemetry messages
+static void send_telemetry_message(void)
+{
+  az_result result;
+
+  if ((result = sample_pnp_thermostat_get_telemetry_message(&client, &sample_thermostat_1, &publish_message))
+      != AZ_OK)
+  {
+    LOG_ERROR("Error getting message and topic for thermostat 1");
+    exit(result);
+  }
+
+  LOG_SUCCESS("Sending Telemetry Message for thermostat 1");
+
+  mqtt_publish_message(publish_message.topic, publish_message.out_payload_span, 0);
+
+  if ((result = sample_pnp_thermostat_get_telemetry_message(&client, &sample_thermostat_2, &publish_message))
+      != AZ_OK)
+  {
+    LOG_ERROR("Error getting message and topic for thermostat 2");
+    exit(result);
+  }
+
+  LOG_SUCCESS("Sending Telemetry Message for thermostat 2");
+
+  mqtt_publish_message(publish_message.topic, publish_message.out_payload_span, 0);
```

Comment:
What about working set telemetry? 
[https://github.com/Azure/opendigitaltwins-dtdl/blob/39ad8b17812e973e69a3605041545e79f0cd34f9/DTDL/v2/samples/TemperatureController.json#L12](https://github.com/Azure/opendigitaltwins-dtdl/blob/39ad8b17812e973e69a3605041545e79f0cd34f9/DTDL/v2/samples/TemperatureController.json#L12)

Reply:
Crap yea. I think I also need the serial number